--- a/asm-util/src/test/java/org/objectweb/asm/util/ASMifierTest.java
+++ b/asm-util/src/test/java/org/objectweb/asm/util/ASMifierTest.java
@@ -107,8 +107,8 @@ public class ASMifierTest extends AsmTes
   }
 
   private static byte[] compile(final String name, final String source) throws IOException {
-    Parser parser = new Parser(new Scanner(name, new StringReader(source)));
     try {
+      Parser parser = new Parser(new Scanner(name, new StringReader(source)));
       UnitCompiler unitCompiler = new UnitCompiler(parser.parseCompilationUnit(), ICLASS_LOADER);
       return unitCompiler.compileUnit(true, true, true)[0].toByteArray();
     } catch (CompileException e) {
